Model/ProductExporter.php

```diff
@@ -0,0 +1,402 @@
+
+    public function __construct(
+        CollectionFactory $productCollectionFactory,
+        Csv $csv,
+        Filesystem $filesystem,
+        DirectoryList $directoryList,
+        File $file,
+        StoreManagerInterface $storeManager,
+        StockRegistryInterface $stockRegistry,
+        TaxCalculationInterface $taxCalculation,
+        Calculation $calculation,
+        TaxClassRepositoryInterface $taxClassRepository,
+        RateCollectionFactory $rateCollectionFactory,
+        ResourceCalculation $resourceCalculation,
+        TaxCalculationCollectionFactory $taxCalculationCollectionFactory,
+        TaxRateRepositoryInterface $taxRateRepository,
+        ProductHelper $productHelper,
+        CategoryRepositoryInterface $categoryRepository,
+        SetFactory $attributeSetFactory,
+        ImageFactory $imageFactory,
+        DateTime $dateTime
+    ) {
+        $this->productCollectionFactory = $productCollectionFactory;
+        $this->csv = $csv;
+        $this->filesystem = $filesystem;
+        $this->directoryList = $directoryList;
+        $this->file = $file;
+        $this->storeManager = $storeManager;
+        $this->stockRegistry = $stockRegistry;
+        $this->taxCalculation = $taxCalculation;
+        $this->calculation = $calculation;
+        $this->taxClassRepository = $taxClassRepository;
+        $this->rateCollectionFactory = $rateCollectionFactory;
+        $this->resourceCalculation = $resourceCalculation;
+        $this->taxCalculationCollectionFactory = $taxCalculationCollectionFactory;
+        $this->taxRateRepository = $taxRateRepository;
+        $this->productHelper = $productHelper;
+        $this->categoryRepository = $categoryRepository;
+        $this->attributeSetFactory = $attributeSetFactory;
+        $this->imageFactory = $imageFactory;
+        $this->dateTime = $dateTime;
+    }
+
+    /**
+     * @throws \Magento\Framework\Exception\FileSystemException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @return void
+     */
+    public function exportProductToCsv(): void
+    {
+        $filePath = $this->directoryList->getPath(DirectoryList::VAR_EXPORT) . $this->getFileName();
+
+        $products = $this->productCollectionFactory->create();
+        $products->addAttributeToSelect('*');
+
+        $data = [
+            self::HEADERS_PATHS,
+            self::HEADERS_LABELS
+        ];
+
+        foreach ($products as $product) {
+            /** @var ProductInterface $product */
+            $productPrice = $product->getPrice();
+            $productSpecialPrice = $product->getSpecialPrice();
+            $productCostPrice = $product->getCost();
+
+            $productData = $this->getProductData($product);
+            $stockData = $this->getStockData($product);
+            $taxData = $this->getTaxData($product);
+            $categoryData = $this->getCategoryData($product);
+
+            $data[] = [
+                $product->getTypeId(),
```

Comment:
there should be some conversion, it's not 1-1 to storekeeper types

Reply:
@ikulis So simple is simple, configurable is configurable, but what to do with bundle, and virtual for example (probably ignore this type, as well as all except simple, config and bundle, because there a bunch of custom product types that 3rd party modules do integrate)?